api/src/main/java/io/opentelemetry/trace/SpanContext.java

```diff
@@ -141,12 +147,16 @@ public boolean isValid() {
    * @return {@code true} if the {@code SpanContext} was propagated from a remote parent.
    * @since 0.1.0
    */
-  public boolean isRemote() {
-    return isRemote;
-  }
+  public abstract boolean isRemote();
 
+  /**
+   * Note: This has a custom equals/hashcode implementation. The equals() method does not include
+   * the {@code tracestate}.
+   *
+   * <p>TODO: is this intentional? If it is, it should be documented as to *why* this is the case.
```

Comment:
+1

Reply:
I suspect that this was not intentional, and fields were added without considering the equals/hashcode. BUT, there are tests that test the equals method [this is weird to start with], that explicitly test the equals not including the tracestate, so I can't be 100% sure. 

It looks like this is vestige of the initial opencensus-based implementation, so maybe @bogdandrutu has some context here?

Reply:
Ping @bogdandrutu  ;)

Reply:
Was just a mistake

Reply:
ok, i'll get rid of it and fix up the tests that break as a result.

Reply:
removed, and removed silly tests for generated equals and toString